conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/RuleValidators.scala

```diff
@@ -54,4 +58,42 @@ object RuleValidators {
       throw new ValidationException(s"$datasetName - $message $errorMessages" )
     }
   }
+
+  /**
+    * Checks if Spark allows casting from one specific type to another.
+    * Throws an instance of ValidationException exception otherwise.
+    *
+    * @param ruleName        A name of a conformance rule to be used as part of an exception error message
+    * @param inputColumnName A name of an input column to be used as part of an exception error message
+    * @param inputType       A type of an input field
+    * @param outputType      A type to cast to
+    * @param spark           (implicit) A Spark Session
+    */
+  @throws[ValidationException]
+  def validateTypeCompatibility(ruleName: String,
+                                inputColumnName: String,
+                                inputType: DataType,
+                                outputType: String)
+                               (implicit spark: SparkSession): Unit = {
+    import spark.implicits._
+
+    // This generates a dataframe we can use to check cast()
+    // Various values need to be here, otherwise Spark may accept incompatible types
+    val df = List("", "true", "false", "0", "1", "-999", "2.2", "a",
+      "2019-01-01", "2019-01-01 00:00:00", "1970-01-01 00:00:00").toDF("dummy")
+
+    // Check if inputType can be cast to outputType
+    // If types are completely incompatible Spark throws an exception
+    try {
+      // The initial type is 'string'. Convert it to 'inputType' first (conversion from string never throws), and than
```

Comment:
Note to myself - conversion from `String` never throws. Interesting. I guess it then returns `null` in the column value,
Cool trick 😎

Reply:
Yes, exactly